### PR TITLE
feat: Improve hover text for common kinds of symbols

### LINKED ISSUE #53: Update pretty-printing of signatures in hover messages to mimic the formatting in hovers from the TypeScript language server
tsserver includes the name of the symbol and a keyword to denote the syntax kind (const/class/interface/...)

CleanShot 2022-03-23 at 15 44 21
CleanShot 2022-03-23 at 15 38 48

### PR DESCRIPTION
**Pull request by varungandhi-src**
Fixes #53. This is a very simplified version of the logic present in the TypeScript codebase.

Enum members do not work currently because of #93. I'll fix that first.

### Test plan

Did some manual testing. Here's a truncated LSIF diff for lsif-node:

```
--- dump-before.lsif	2022-04-27 16:10:17.000000000 -0700
+++ dump-after.lsif	2022-04-27 16:59:10.000000000 -0700
@@ -5 +5 @@
-{"id":5,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nWriter\n```\n\n---\n\n"}}}
+{"id":5,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\ninterface Writer\n```\n\n---\n\n"}}}
@@ -16 +16 @@
-{"id":16,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(...data: string[]) => void\n```\n\n---\n\n"}}}
+{"id":16,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(method) writeln(...data: string[]) => void\n```\n\n---\n\n"}}}
@@ -26 +26 @@
-{"id":26,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nstring[]\n```\n\n---\n\n"}}}
+{"id":26,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(parameter) data: string[]\n```\n\n---\n\n"}}}
@@ -36 +36 @@
-{"id":36,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nFileWriter\n```\n\n---\n\n"}}}
+{"id":36,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nclass FileWriter\n```\n\n---\n\n"}}}
@@ -46 +46 @@
-{"id":46,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nnumber\n```\n\n---\n\n"}}}
+{"id":46,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(property) fd: number\n```\n\n---\n\n"}}}
@@ -56 +56 @@
-{"id":56,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(...data: string[]) => void\n```\n\n---\n\n"}}}
+{"id":56,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(method) writeln(...data: string[]) => void\n```\n\n---\n\n"}}}
@@ -66 +66 @@
-{"id":66,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nstring[]\n```\n\n---\n\n"}}}
+{"id":66,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(parameter) data: string[]\n```\n\n---\n\n"}}}
@@ -76 +76 @@
-{"id":76,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(buffer: Buffer) => void\n```\n\n---\n\n"}}}
+{"id":76,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(method) writeBuffer(buffer: Buffer) => void\n```\n\n---\n\n"}}}
@@ -86 +86 @@
-{"id":86,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\nBuffer\n```\n\n---\n\n"}}}
+{"id":86,"type":"vertex","label":"hoverResult","result":{"contents":{"kind":"markdown","value":"```ts\n(parameter) buffer: Buffer\n```\n\n---\n\n"}}}
@@ -96 +96 @@
```

I will add automated tests once TJ has ported the hover testing code to TypeScript.

### CODE REVIEW

File: snapshots/output/multi-project/packages/a/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
   export function a(): string {
 //                ^ definition @example/a 1.0.0 src/`index.ts`/a().
-//                documentation ```ts\n() => string\n```
+//                documentation ```ts\nfunction a() => string\n```
```
Comment (olafurpg) [inline snapshots/output/multi-project/packages/a/src/index.ts]: I would expect the syntax to use `:` as a separator for functions
```suggestion
//                documentation ```ts\nfunction a(): string\n```
```

**Review by olafurpg — APPROVED** [review]
LGTM 👍 This is already a big upgrade. I think it would be nice to have a special case for `function` so that we use `:` instead of `=>` but we can fix that in a separate PR as well.

File: snapshots/output/syntax/src/accessors.ts [hunk 2]
```diff
@@ -1,81 +1,81 @@
...
 //      ^^^^^^ definition syntax 1.0.0 src/`accessors.ts`/C#`<get>length`().
-//      documentation ```ts\nnumber\n```
+//      documentation ```ts\nget length: number\n```
       return this._length
 //                ^^^^^^^ reference syntax 1.0.0 src/`accessors.ts`/C#_length.
     }
     set length(value: number) {
 //      ^^^^^^ definition syntax 1.0.0 src/`accessors.ts`/C#`<get>length`().
-//      documentation ```ts\nnumber\n```
+//      documentation ```ts\nget length: number\n```
```
Comment (varungandhi-src) [inline snapshots/output/syntax/src/accessors.ts:15]: This is incorrect.

Comment (varungandhi-src) [reply snapshots/output/syntax/src/accessors.ts:15]: Hmm, there seems to be something consistently wrong with how we're handling accessors. Notice the

```
definition syntax 1.0.0 src/`accessors.ts`/C#`<get>length`().
``` 

right above this, it should be `<set>` and not `<get>`. I'll fix the accessors in a separate PR.

**Pull request merged**